wcfsetup/install/files/lib/system/form/option/formatter/CurrencyFormatter.class.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace wcf\system\form\option\formatter;
+
+use wcf\system\language\LanguageFactory;
+use wcf\util\StringUtil;
+
+/**
+ * Formatter for currencies.
+ *
+ * @author      Marcel Werk
+ * @copyright   2001-2025 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.2
+ */
+class CurrencyFormatter implements IFormOptionFormatter
+{
+    #[\Override]
+    public function format(string $value, int $languageID, array $configurationData): string
+    {
+        $showDecimals = $value % 100 != 0;
```

Comment:
```suggestion
        $showDecimals = $value % 100 !== 0;
```

Besides, I’m not sure if this is a good decision at all. Having fixed decimal places is common for currencies because now you end up `99.99`, `100` and `100.01` mixed together?